File: test/perf.rb
require "rubygems"
# gem install json
require "json"
# gem install json_pure
require "json/pure"
# gem install yajl-ruby
require "yajl"

require File.join(File.dirname(__FILE__), "json_samples")

def time(name, &block)
  elapsed = (1..5).map {|i| s = Time.now; block.call; (Time.now - s) * 1000 }.min
  puts name+"\t"+elapsed.to_s
end

puts "Ruby Performance (msecs)"

time "JSON::Ext" do
  JSON::Ext::Parser.new($a).parse
  JSON::Ext::Parser.new($b).parse
  JSON::Ext::Parser.new($c).parse
  JSON::Ext::Parser.new($d).parse
  JSON::Ext::Parser.new($e).parse
end

time "JSON::Pure" do
  JSON::Pure::Parser.new($a).parse
  JSON::Pure::Parser.new($b).parse
  JSON::Pure::Parser.new($c).parse
  JSON::Pure::Parser.new($d).parse
  JSON::Pure::Parser.new($e).parse
end

time "Yajl\t" do
  Yajl::Parser.parse($a)
  Yajl::Parser.parse($b)
  Yajl::Parser.parse($c)
  Yajl::Parser.parse($d)
  Yajl::Parser.parse($e)
end

puts ""

File: test/perf.jl
require("JSON")

require("JSON/test/json_samples")

macro timeit(name, ex)
  quote
    t = Inf
    for i=1:1000
      t = min(t, @elapsed $ex)
    end
    println($name, "\t", JSON.Faster.format(t * 1000))
  end
end

println("Julia Performance (msecs)")

@timeit "JSON\t" begin
  @assert JSON.parse(a) != nothing
  @assert JSON.parse(b) != nothing
  @assert JSON.parse(c) != nothing
  @assert JSON.parse(d) != nothing
  @assert JSON.parse(e) != nothing
end

@timeit "FasterJSON" begin
  @assert JSON.Faster.parse(a) != nothing
  @assert JSON.Faster.parse(b) != nothing
  @assert JSON.Faster.parse(c) != nothing
  @assert JSON.Faster.parse(d) != nothing
  @assert JSON.Faster.parse(e) != nothing
end

println()
